src/core/slice.c

```diff
@@ -347,6 +348,95 @@ static void slice_enumerate_perpetual(Manager *m) {
                 (void) slice_make_perpetual(m, SPECIAL_SYSTEM_SLICE, NULL);
 }
 
+static int slice_freezer_action_supported_by_children(Unit *s) {
+        int r;
+        _cleanup_closedir_ DIR *d = NULL;
+
+        assert(s);
+
+        r = cg_enumerate_subgroups(SYSTEMD_CGROUP_CONTROLLER, s->cgroup_path, &d);
+        if (r < 0)
+                return r;
+
+        for (;;) {
+                _cleanup_free_ char *subgroup = NULL;
+                Unit *u;
+
+                r = cg_read_subgroup(d, &subgroup);
+                if (r < 0)
+                        return r;
+                if (r == 0)
+                        break;
+
+                u = manager_get_unit(s->manager, subgroup);
+                assert(u);
+
+                if (u->type == UNIT_SLICE) {
+                        r = slice_freezer_action_supported_by_children(u);
+                        if (r <= 0)
+                                return r;
+                } else if (!UNIT_VTABLE(u)->freeze || !UNIT_VTABLE(u)->thaw)
```

Comment:
I see that `unit_freezer_action` does some logical checks on freezing suitability (e.g. no running job), freezing via slice seems to bypass this conditions.